### PR TITLE
Gsoc2016/temp

### PR DESCRIPTION
**Pull request by vladbat00**

### CODE REVIEW

File: autotests/plugins/cli7zplugin/cli7ztest.cpp [hunk 1]
```diff
@@ -379,3 +380,56 @@ void Cli7zTest::testExtractArgs()
...
+    QVERIFY(plugin);
+
+    QFETCH(QList<Archive::Entry*>, files);
+    QFETCH(Archive::Entry*, destination);
+
+    CompressionOptions options = CompressionOptions();
+    options.insert(QStringLiteral("GlobalWorkDir"), QFINDTESTDATA("data"));
+    AddJob *addJob = new AddJob(files, destination, options, plugin);
+    TestHelper::startAndWaitForResult(addJob);
+    addJob->deleteLater();
```
Comment (elvisangelaccio) [inline autotests/plugins/cli7zplugin/cli7ztest.cpp:429]: Don't delete the jobs manually. Same for all the instances in the other tests.

File: autotests/plugins/libarchiveplugin/libarchivetest.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+QTEST_GUILESS_MAIN(LibarchiveTest)
+
+using namespace Kerfuffle;
+
+void LibarchiveTest::initTestCase()
+{
+    m_plugin = new Plugin(this);
+    foreach (Plugin *plugin, m_pluginManger.availablePlugins()) {
+        if (plugin->metaData().pluginId() == QStringLiteral("kerfuffle_cli7z")) {
```
Comment (elvisangelaccio) [inline autotests/plugins/libarchiveplugin/libarchivetest.cpp:40]: You probably are looking for `kerfuffle_libarchive` here (or `kerfuffle_libarchive_readonly`)

File: kerfuffle/jobs.h [hunk 3]
```diff
@@ -224,7 +224,7 @@ class KERFUFFLE_EXPORT AddJob : public Job
     Q_OBJECT
 
 public:
-    AddJob(QList<Archive::Entry*> &files, const CompressionOptions& options, ReadWriteArchiveInterface *interface);
+    AddJob(const QList<Archive::Entry*> &files, const Archive::Entry *destination, const CompressionOptions& options, ReadWriteArchiveInterface *interface);
```
Comment (elvisangelaccio) [inline kerfuffle/jobs.h:227]: If you set a default `destination` here, you don't need to pass `new Archive::Entry(this)` all over the place (when you don't want to add files to a specific destination).

File: kerfuffle/cliinterface.cpp [hunk 4]
```diff
@@ -164,14 +164,56 @@ bool CliInterface::copyFiles(const QList<Archive::Entry*> &files, const QString
...
+                preservedParent = file->parent();
+            }
+
+            const QString filePath = QDir::currentPath() + QLatin1Char('/') + file->property("fullPath").toString();
+            QString newFilePath = absoluteDestinationPath + file->property("fullPath").toString();
+            // Symlink function can't accept the second argument as a path with trailing slash.
+            if (newFilePath[newFilePath.count() - 1] == QLatin1Char('/')) {
+                newFilePath.remove(newFilePath.count() - 1, 1);
+            }
+            if (symlink(filePath.toStdString().c_str(), newFilePath.toStdString().c_str()) != 0) {
```
Comment (elvisangelaccio) [inline kerfuffle/cliinterface.cpp:199]: I'm not sure `symlink` is portable. Can you use `QFile::link()` here?

File: kerfuffle/jobs.h (hunk 3 above)
Comment (vladbat00) [inline kerfuffle/jobs.h:227]: Do you suggest using Q_NULLPTR?

File: autotests/plugins/libarchiveplugin/libarchivetest.cpp (hunk 2 above)
Comment (vladbat00) [inline autotests/plugins/libarchiveplugin/libarchivetest.cpp:40]: Yeah, thanks, forgot to fix this copypaste)

File: kerfuffle/jobs.h (hunk 3 above)
Comment (elvisangelaccio) [inline kerfuffle/jobs.h:227]: Yeah, try and see whether it's doable. Otherwise drop this comment ;)

**Pull request closed without merging**